src/dune_rules/buildable_rules.ml

```diff
@@ -135,6 +137,29 @@ let modules_rules
         fun name -> default || List.mem executable_names name ~equal:Module_name.equal)
       else fun _ -> default
   in
+  let* () =
+    match ctypes with
+    | Some ctypes ->
+      let (ctypes : Ctypes_field.t) = ctypes in
+      let modules = Modules.With_vlib.modules modules in
+      (ctypes.type_description.functor_loc, ctypes.type_description.functor_)
+      :: List.map
+           ~f:(fun (x : Ctypes_field.Function_description.t) -> x.functor_loc, x.functor_)
+           ctypes.function_description
+      |> Memo.parallel_iter ~f:(fun ((functor_loc, m) : Loc.t * Module_name.t) ->
```

Comment:
Why not just `List.iter`? Also, that `List.map` is unnecesasry.

Reply:
If we don't use the parallel iter, then we can't get all the error messages at once. I think it is more useful for users to see all the errors rather than one at a time.

Reply:
The `tpye_description` and `function_description` records are different but they have similar fields. We need to pick the fields that we wish to act on. Therefore the List.map becomes necessary. If you know of a better way to do that, let me know.

Reply:
I see. You should leave a comment to explain that you're relying on `parallel_iter` to raise more than one error. Raising multiple errors is a fiber primitive, it's a shame we need to it in a roundabout way here.